### DISCUSSION TITLE
Can i get darker hex color for given hex color?

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [ImageMagick/ImageMagick#5225] [Help] [answered]
Hi to all!

I have #f4bd45(for example) color? can i get darker or lighter color of this in percentage with Magick.Net?

**Comment by snibgo** [comment]
I don't understand the question. If you know the channel values of a colour, you can make a lighter or darker version of that hue by multiplying the channels values by the same number.

**Comment by fmw42** [comment]
I do not understand the question. A hex value is fixed. There is no other value for it. If you want a darker color, then the hex value would need to change. If you want the hex color converted to a percentage, you would have to use rgb(r%,g%,b%) format for the color. If you want the same hue, but a darker color, you can use -modulate to change the colors.

**Comment by jaka1-newbie** [comment]
I need new color, based at given hex code.
-modulate operator change whole image. I need only color.

**Reply by fmw42** [reply]
`convert image -fuzz XX% -fill "newcolor" -opaque "oldcolor" result`

set XX% as desired to allow similar colors. If XX=0, then it will find and change only the one oldcolor to the newcolor. Colors can be names, hex or rgb() colors.

If on Imagemagick 7, change convert to magick

**Comment by jaka1-newbie** [comment]
I use Magick.Net
And this color has not yet appeared in the image. 
I think a need convert from hex to hsl and change brightness, after this convert to hex

**Comment by jaka1-newbie** [comment]
Didnt find examples yet(

**Comment by jaka1-newbie** [comment]
```
Color color = ColorTranslator.FromHtml("#FFAAFF");
int r = Convert.ToInt16(color.R);
int g = Convert.ToInt16(color.G);
int b = Convert.ToInt16(color.B);
var l = color.GetBrightness();
```
Got Brightness, dont understand how change it

**Comment by jaka1-newbie** [comment]
```
Color color = ColorTranslator.FromHtml("#f4bd45");
Color c2 = Color.FromArgb(color.A,
        (int)(color.R * 0.8), (int)(color.G * 0.8), (int)(color.B * 0.8));
string hex = c2.R.ToString("X2") + c2.G.ToString("X2") + c2.B.ToString("X2");
```

**Comment by jaka1-newbie** [comment]
this is solution

**Comment by dlemstra** [comment, marked as answer]
In the next version of Magick.NET you will be able to do this: `var color = new MagickColor("#f4bd45") * new Percentage(80);`
